compiler/rustc_resolve/src/lib.rs

```diff
@@ -1808,6 +1809,12 @@ impl<'a, 'tcx> Resolver<'a, 'tcx> {
                     }
                 }
             }
+            if let ImportKind::MacroUse { warn_not_exported } = import.kind
```

Comment:
Could you put this block above the `if is_lexical_scope { ... }` logic, so it's not skipped if return happens?
I'm not sure if it makes any difference, but just in case.